src/twisted/python/_url.py

```diff
@@ -7,787 +7,7 @@
 URL parsing, construction and rendering.
 """
 
-try:
-    from urlparse import urlsplit, urlunsplit
-    from urllib import quote as urlquote, unquote as urlunquote
-except ImportError:
-    from urllib.parse import (urlsplit, urlunsplit,
-                              quote as urlquote,
-                              unquote_to_bytes as urlunquote)
+from hyperlink._url import URL
```

Comment:
Why are we importing from a private API ?

I think that hyperlink should make a vow and expose a public API :)